ex1/dates_times.py

```diff
@@ -32,13 +32,31 @@
 
 # Define a function named last_of_month that takes an argument dt of type date
 # and returns a date object representing the last day of the month dt was in.
+import datetime
 
+def last_of_month(dt):
+    return(dt.replace(month = dt.month + 1, day = 1) - datetime.timedelta(days = 1))
```

Comment:
`calendar.monthrange` can do the work for you. See for example [here](https://github.com/TUW-GEO-python-intro/Exercise-1/pull/20/files#diff-b5dc1ec9f5b885f64b8dbe958d8d4a56R54)